core/client/fs/src/main/java/alluxio/client/file/dora/DoraCacheFileInStream.java

```diff
@@ -27,23 +27,19 @@
  * Implementation of {@link FileInStream} that reads from a dora cache if possible.
  */
 public class DoraCacheFileInStream extends FileInStream {
-
-  private final DataReader.Factory mReaderFactory;
   private final long mLength;
-
   private long mPos = 0;
   private boolean mClosed;
-  private DataReader mDataReader;
-  private DataBuffer mCurrentChunk;
+  private final DoraDataReader mDataReader;
 
   /**
    * Constructor.
-   * @param readerFactory
+   * @param reader
    * @param length
    */
-  public DoraCacheFileInStream(DataReader.Factory readerFactory,
+  public DoraCacheFileInStream(DoraDataReader reader,
```

Comment:
DoraCacheFileInStream can internally have DoraCachePositionReader so that the position read logics can be shared.
I will work on this part later